From 94bc0b33d7ec7677507ae9fa311e3bce0bd107de Mon Sep 17 00:00:00 2001
From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Date: Mon, 7 Oct 2024 10:45:13 +0300
Subject: [PATCH 08/14] media/gpu: Enable v4l2 encoding support on Linux

v4l2 encoding is currently enable only on ChromeOS because it depends on
ChromeOS specific v4l2 driver behavior to handle some kinds of dmabufs
(crbug.com/901264). For many use cases, though, dmabufs are not even
used by default (e.g., webrtc), so enable this to take advantage of
accelerated encoding for many common cases.
---
 media/gpu/gpu_video_encode_accelerator_factory.cc | 2 +-
 media/gpu/v4l2/BUILD.gn                           | 7 ++-----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/media/gpu/gpu_video_encode_accelerator_factory.cc b/media/gpu/gpu_video_encode_accelerator_factory.cc
index 0345fc65f7..3a7d8a527f 100644
--- a/media/gpu/gpu_video_encode_accelerator_factory.cc
+++ b/media/gpu/gpu_video_encode_accelerator_factory.cc
@@ -47,7 +47,7 @@ namespace media {
 namespace {
 #if BUILDFLAG(USE_V4L2_CODEC)
 std::unique_ptr<VideoEncodeAccelerator> CreateV4L2VEA() {
-#if BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_LINUX)
   // TODO(crbug.com/901264): Encoders use hack for passing offset within
   // a DMA-buf, which is not supported upstream.
   return base::WrapUnique<VideoEncodeAccelerator>(
diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index bd5d25000c..5ab8675c79 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -51,6 +51,8 @@ source_set("v4l2") {
     "v4l2_video_decoder_delegate_vp8.h",
     "v4l2_video_decoder_delegate_vp9.cc",
     "v4l2_video_decoder_delegate_vp9.h",
+    "v4l2_video_encode_accelerator.cc",
+    "v4l2_video_encode_accelerator.h",
     "v4l2_vp9_helpers.cc",
     "v4l2_vp9_helpers.h",
   ]
@@ -75,11 +77,6 @@ source_set("v4l2") {
       # AV1 delegate depends on header files only in ChromeOS SDK
       "v4l2_video_decoder_delegate_av1.cc",
       "v4l2_video_decoder_delegate_av1.h",
-
-      # TODO(crbug.com/901264): Encoders use hack for passing offset
-      # within a DMA-buf, which is not supported upstream.
-      "v4l2_video_encode_accelerator.cc",
-      "v4l2_video_encode_accelerator.h",
     ]
   }
 
-- 
2.45.2

